app/assets/javascripts/camps.js

```diff
@@ -27,6 +28,67 @@ $(document).ready(function() {
   $('.budgetprogressbar').budgetprogressbar();
 
   Filterrific.init();
+
+  function getAuthenticityToken(el) {
+    return $('input[name=authenticity_token]', $(el).closest('form')).val()
+  }
+
+  function getUrl(el) {
+    return $(el).closest('form').attr('action');
+  }
+
+  function refreshTags(data) {
+    $('#tag-container').empty();
+    for(var tag in data) {
+      $('#tag-container').append(
+          '<div class="tag-item">'
+          + data[tag].name
+          + '<span class="tag-remove" data-tag="'
+          + data[tag].name
+          + '">x</span>'
+          + '</div>'
+      );
+      $('.tag-remove').click(removeTag);
+    }
+  }
+
+  function removeTag(e, el) {
+    var url = getUrl(e.target);
+    var tag = $(e.target).data('tag');
+    var token = getAuthenticityToken(e.target);
+
+    $.ajax({
+      url: url,
+      type: 'POST',
+      data: {method:'_patch', 'camp[tag]': tag, authenticity_token:token},
+      success: function(data) {
+        refreshTags(data);
+      }
+    })
+  }
+
+  function addTags(e, el) {
+    var url = getUrl(e.target);
+    var tags = $("#camp_tag_list").val();
+    var token = getAuthenticityToken(e.target);
+
+    e.preventDefault();
+
+    $.ajax({
+      url: url,
+      type: 'POST',
+      data: {method:'_patch', 'camp[tag_list]': tags, authenticity_token:token},
+      success: function(data) {
+        refreshTags(data);
+      }
+    });
+
+    return false;
+  }
+
+  $('.tag-remove').click(removeTag);
```

Comment:
Rather than adding all this javascript, I'd highly recommend looking at using rails's UJS to do posts from hyperlinks:
https://github.com/rails/jquery-ujs/wiki

Reply:
I did see UJS. In this case we chose to build on existing infrastructure and scripts that are allready loaded on the page instead of introducing new scripts and libraries. How strongly do you feel about this? Would it be worth it to refactor all our current javascript to UJS?